tests/integration/test_firehose.py

```diff
@@ -239,3 +239,62 @@ def assert_s3_contents():
         finally:
             firehose_client.delete_delivery_stream(DeliveryStreamName=delivery_stream_name)
             es_client.delete_elasticsearch_domain(DomainName=domain_name)
+
+    def test_delivery_stream_with_kinesis_as_source(
+        self,
+        firehose_client,
+        kinesis_client,
+        s3_bucket,
+        kinesis_create_stream,
+    ):
+
+        bucket_arn = aws_stack.s3_bucket_arn(s3_bucket)
+        stream_name = f"test-stream-{short_uid()}"
+        log_group_name = f"group{short_uid()}"
+        role_arn = "arn:aws:iam::000000000000:role/Firehose-Role"
+        delivery_stream_name = f"test-delivery-stream-{short_uid()}"
+
+        kinesis_create_stream(StreamName=stream_name, ShardCount=2)
+        stream_arn = kinesis_client.describe_stream(StreamName=stream_name)["StreamDescription"][
+            "StreamARN"
+        ]
+
+        response = firehose_client.create_delivery_stream(
+            DeliveryStreamName=delivery_stream_name,
+            DeliveryStreamType="KinesisStreamAsSource",
+            KinesisStreamSourceConfiguration={
+                "KinesisStreamARN": stream_arn,
+                "RoleARN": role_arn,
+            },
+            ExtendedS3DestinationConfiguration={
+                "BucketARN": bucket_arn,
+                "RoleARN": role_arn,
+                "BufferingHints": {"IntervalInSeconds": 60, "SizeInMBs": 64},
+                "DynamicPartitioningConfiguration": {"Enabled": True},
+                "ProcessingConfiguration": {
+                    "Enabled": True,
+                    "Processors": [
+                        {
+                            "Type": "MetadataExtraction",
+                            "Parameters": [
+                                {
+                                    "ParameterName": "MetadataExtractionQuery",
+                                    "ParameterValue": "{s3Prefix: .tableName}",
+                                },
+                                {"ParameterName": "JsonParsingEngine", "ParameterValue": "JQ-1.6"},
+                            ],
+                        },
+                    ],
+                },
+                "DataFormatConversionConfiguration": {"Enabled": True},
+                "CompressionFormat": "GZIP",
+                "Prefix": "firehoseTest/!{partitionKeyFromQuery:s3Prefix}/!{partitionKeyFromLambda:companyId}/!{partitionKeyFromLambda:year}/!{partitionKeyFromLambda:month}/",
+                "ErrorOutputPrefix": "firehoseTest-errors/!{firehose:error-output-type}/",
+                "CloudWatchLoggingOptions": {
+                    "Enabled": True,
+                    "LogGroupName": log_group_name,
+                },
+            },
+        )
+
```

Comment:
Let's add a cleanup, i.e., deleting the created delivery stream, at the end of the test method.